examples/dataflow-flex-python/README.md

```diff
@@ -0,0 +1,202 @@
+export LOCATION=us-central1
+export BQ_DATASET=bqtoxmldataset
+export NETWORK=shared-vpc
+export SUBNET=bh-subnet-usc1
+export REPO=dataflowf-image-repo
+export DF_WORKER_SA=dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com
+
+```
+
+#### Setup IAM
+```
+# Create service account for dataflow workers and launchers
+gcloud iam service-accounts create dataflow-worker-sa --project=$PROJECT_ID
+
+# Assign dataflow worker permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/dataflow.worker
+
+# Assign Object viewer permissions in order to read the data from cloud storage
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/storage.objectViewer
+
+# Assign Object viewer permissions in order to Create temp files cloud storage
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/storage.objectCreator
+
+# Assign Service Account User permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/iam.serviceAccountUser
+
+# Assign BigQuery job ser permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.jobUser
+
+# Assign bigquery data editor permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.dataEditor
+
+# Assign Artifactory Reader Permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/artifactregistry.reader
+
+# Assign network user permissions on Host project
+gcloud projects add-iam-policy-binding $HOST_PROJECT_ID --member "serviceAccount:service-$PROJECT_NUMBER@dataflow-service-producer-prod.iam.gserviceaccount.com" --role roles/compute.networkUser
+```
+
+#### Setup Cloud Storage
+```
+# Create Cloud Storage bucket for input data
+gcloud storage buckets create gs://$INPUT_BUCKET_NAME --location $LOCATION --project $PROJECT_ID
+
+# Create a bucket for dataflow staging and temp locations
+gcloud storage buckets create gs://$STAGING_BUCKET_NAME --location $LOCATION --project $PROJECT_ID
+
+gsutil iam ch serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com:roles/storage.legacyBucketWriter gs://$STAGING_BUCKET_NAME
+```
+
+#### Create BQ Dataset
+```
+bq --location=us-central1 mk --dataset $PROJECT_ID:$BQ_DATASET
+```
+
+#### Create ARtifactory registry
+```
+gcloud  artifacts  repositories create $REPO --location $LOCATION --repository-format docker --project $PROJECT_ID
+```
+
+### Build and Push Docker Images for template
```

Comment:
I would split steps between prerequisites (up to this one) and tasks relevant to the example to make it more readable for the user.

Reply:
done